site/docs/plugins/inline-query.md

```diff
@@ -0,0 +1,267 @@
+await ctx.answerInlineQuery(results, { button });
+```
+
+When the user presses the button, a `/start` command message will be sent to your bot.
+The start parameter will be available via [deep linking](../guide/commands.md#deep-linking-support).
+In other words, using the above code snippet, `ctx.match` will have the value `"login"` in your command handler.
+
+If you then send an [inline keyboard](./keyboard.md#building-a-custom-keyboard) with a `switchInline` button, the user will be returned to the chat where they pressed the inline query results button initially.
+
+```ts
+bot
+  .command("start")
+  .filter((ctx) => ctx.match === "login", async (ctx) => {
+    // User is coming from inline query results
+    await ctx.reply("DM open, you can go back now!", {
+      reply_markup: new InlineKeyboard()
+        .switchInline("Go back"),
+    });
+  });
+```
+
+That way, you can perform e.g. login procedures in a private chat with the user before delivering inline query results.
+The dialogue can go back and forth a bit before you send them back.
+For example, you can [enter a short conversation](./conversations.md#installing-and-entering-a-conversation) with the conversations plugin.
+
+## Getting Feedback About Chosen Results
+
+Inline query results are delivered in a fire-and-forget fashion.
+In other words, after your bot sent the list of inline query results to Telegram, it will not know which result the user picked (or if they picked one at all).
+
+If you are interested in this, you can enable inline feedback with [@BotFather](https://t.me/grammyjs).
+You can decide how much feedback you want to receive by chosing among several options between 0 % (feedback disabled) and 100 % (receive feedback for every chosen inline result).
+
+Inline feedback is delivered via `chosen_inline_result` updates.
+You can listen for specific result identifiers via string or regular expression.
+Naturally, you can also listen for the updates the normal way via filter queries.
+
+```ts
+// Listen for specific result identifiers.
+bot.chosenInlineResult(/id-[0-9]+/, async (ctx) => {
+  const match = ctx.match; // regex match object
+  const query = ctx.chosenInlineResult.query; // used inline query
+});
+
+// Listen for any chosen inline results.
+bot.on("chosen_inline_result", async (ctx) => {
+  const query = ctx.chosenInlineResult.query; // used inline query
+});
+```
+
+Some bots set the feedback to 100 % and use it as a hack.
+They deliver dummy messages with no real content in `answerInlineQuery`.
+Immediately after receiving a `chosen_inline_result` update, they edit the respective message and inject the real message content.
+
+These bots will not work for anonymous admins or when sending scheduled messages, as no inline feedback can be received there.
+However, if this is not a problem for you, then this hack will allow you to not have to generate a lot of message content for messages that never end up being sent.
+This can save your bot resources.
```

Comment:
```suggestion
This can save your resources.
```

Reply:
This doesn't make sense. It saves the resources of the bot. It's like saying "this will save you time" just that it's about your bot not you.

Reply:
Alright!